NET-6821 Disable Terminating Gateway Auto Host Header Rewrite (#20802)
hashicorp/consul: website/content/docs/connect/config-entries/terminating-gateway.mdx

          `An optional hostname or domain name to specify during the TLS handshake. This option will also configure [strict SAN matching](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#envoy-v3-api-field-extensions-transport-sockets-tls-v3-certificatevalidationcontext-match-typed-subject-alt-names), which requires
            the external services to have certificates with SANs, not having which will result in \`CERTIFICATE_VERIFY_FAILED\` error.`,
        },
        {
          name: 'DisableAutoHostRewrite',
          type: 'bool: ""',
          description:

>             'When set to true, Terminating Gateway will not modify the incoming requests host header for this service.',

Author: ```suggestion
           'When set to true, disables the terminating gateway's ability to modify this service's incoming requests host header.',
```
